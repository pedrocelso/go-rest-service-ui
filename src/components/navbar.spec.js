import React from 'react';
import {mount, shallow} from 'enzyme'
import sinon from 'sinon'

import NavBar from './navbar'

describe(`<NavBar />`, () => {
  it(`Should render without errors`, () => {
    const component = shallow(<NavBar redirect={sinon.fake()} />);

    expect(component).toMatchSnapshot();
  });

  it(`.getTitle() - Should return the right title based on the document location`, () => {
    expect.assertions(2);
    const component = mount(<NavBar redirect={sinon.fake()} />);;

    expect(component.instance().getTitle(`/users`)).toEqual(`Users`)
    expect(component.instance().getTitle(`/tasks`)).toEqual(`Tasks`)
  });

  it(`.getTitle() - Should return an empty string for unknow locations`, () => {
    expect.assertions(1);
    const component = mount(<NavBar redirect={sinon.fake()} />);;

    expect(component.instance().getTitle(`/gerere`)).toEqual(``)
  });

  it(`.toggleDrawer() - Should set the right parameter for the drawer menu`, () => {
    const component = mount(<NavBar redirect={sinon.fake()} />);;
    expect(component.state().drawerOpened).toBeFalsy();
    component.instance().toggleDrawer(true)();
    expect(component.state().drawerOpened).toBeTruthy();
    component.instance().toggleDrawer(false)();
    expect(component.state().drawerOpened).toBeFalsy();
  });
});