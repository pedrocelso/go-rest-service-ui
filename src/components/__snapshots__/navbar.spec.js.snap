// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> .render() - Should render the create button on tasks 1`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Menu"
        className="menu-button"
        color="inherit"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="toolbar-title"
        color="inherit"
        variant="h6"
      >
        Tasks
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button))
        color="secondary"
        variant="contained"
      >
        Create
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    onClose={[Function]}
    open={false}
    variant="temporary"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      key="Users"
      onClick={[Function]}
      to="/users"
    >
      <PeopleIcon
        color="primary"
      />
      <WithStyles(ForwardRef(ListItemText))
        primary="Users"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      key="Tasks"
      onClick={[Function]}
      to="/tasks"
    >
      <StorageIcon
        color="primary"
      />
      <WithStyles(ForwardRef(ListItemText))
        primary="Tasks"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(Divider))
      key="divider"
    />
    <WithStyles(ForwardRef(ListItem))
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      key="Logout"
      onClick={[Function]}
      to="/logout"
    >
      <LockOpenIcon
        color="primary"
      />
      <WithStyles(ForwardRef(ListItemText))
        primary="Logout"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(Drawer))>
</div>
`;

exports[`<NavBar /> .render() - Should render the create button on tasks 2`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Menu"
        className="menu-button"
        color="inherit"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="toolbar-title"
        color="inherit"
        variant="h6"
      >
        Login
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    onClose={[Function]}
    open={false}
    variant="temporary"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      key="Login"
      onClick={[Function]}
      to="/"
    >
      <LockIcon
        color="primary"
      />
      <WithStyles(ForwardRef(ListItemText))
        primary="Login"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(Divider))
      key="divider"
    />
  </WithStyles(ForwardRef(Drawer))>
</div>
`;

exports[`<NavBar /> Should render without errors for authenticated users 1`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Menu"
        className="menu-button"
        color="inherit"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="toolbar-title"
        color="inherit"
        variant="h6"
      >
        Users
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    onClose={[Function]}
    open={false}
    variant="temporary"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      key="Users"
      onClick={[Function]}
      to="/users"
    >
      <PeopleIcon
        color="primary"
      />
      <WithStyles(ForwardRef(ListItemText))
        primary="Users"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      key="Tasks"
      onClick={[Function]}
      to="/tasks"
    >
      <StorageIcon
        color="primary"
      />
      <WithStyles(ForwardRef(ListItemText))
        primary="Tasks"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(Divider))
      key="divider"
    />
    <WithStyles(ForwardRef(ListItem))
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      key="Logout"
      onClick={[Function]}
      to="/logout"
    >
      <LockOpenIcon
        color="primary"
      />
      <WithStyles(ForwardRef(ListItemText))
        primary="Logout"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(Drawer))>
</div>
`;

exports[`<NavBar /> Should render without errors for unauthenticated users 1`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Menu"
        className="menu-button"
        color="inherit"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="toolbar-title"
        color="inherit"
        variant="h6"
      >
        Login
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    onClose={[Function]}
    open={false}
    variant="temporary"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      key="Login"
      onClick={[Function]}
      to="/"
    >
      <LockIcon
        color="primary"
      />
      <WithStyles(ForwardRef(ListItemText))
        primary="Login"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(Divider))
      key="divider"
    />
  </WithStyles(ForwardRef(Drawer))>
</div>
`;
